### PR TITLE
Deduplicate code

### LINKED ISSUE #24: Deduplicate code
Currently there is a lot of duplicated code for things such as live processing of Kafka messages, etc. which should be addressed.

### PR DESCRIPTION
**Pull request by Henry-Denny**
Closes #24.

### CODE REVIEW

File: common/src/lib.rs [hunk 1]
```diff
@@ -44,3 +47,28 @@ pub fn generate_kafka_client_config(
     }
     client_config
 }
+
+pub fn create_default_consumer(
+    broker_address: &String,
+    username: &Option<String>,
+    password: &Option<String>,
+    consumer_group: &String,
+    topics_to_subscribe: Option<&[&str]>,
```
Comment (Modularius) [inline common/src/lib.rs:56]: Does `topics_to_subscribe: Option<&[&str]>` need to be `Option`? What happens when `consumer.subscribe` is given an empty list? I don't think that case will ever arise anyway as a consumer should always subscribe to at least one topic.

Comment (Henry-Denny) [reply common/src/lib.rs:56]: That makes sense, I'm not sure why I made it `Option` to begin with.

File: diagnostics/src/daq_trace/mod.rs [hunk 2]
```diff
@@ -79,6 +79,46 @@ impl DigitiserData {
     }
 }
 
+type SharedData = Arc<Mutex<HashMap<u8, DigitiserData>>>;
```
Comment (DanNixon) [inline diagnostics/src/daq_trace/mod.rs:82]: Are the changes to this file intentional?

Comment (Henry-Denny) [reply diagnostics/src/daq_trace/mod.rs:82]: They are not.

**Review by DanNixon — APPROVED** [review]

**Comment by DanNixon** [comment]
@Henry-Denny There is a conflict here.

**Review by DanNixon — APPROVED** [review]

**Pull request merged**